test/test_quantized.py

```diff
@@ -1055,109 +1055,120 @@ class TestDynamicQuantizedLinear(TestCase):
-            b_fp32 = torch.from_numpy(
-                _dequantize(b_q0, X_scale * int(W_scales[0].item()), 0)
-            ).to(dtype=torch.float) if use_bias else None
+            avoid_vpmaddubsw_overflow_linear(
+                batch_size,
+                input_channels,
+                output_channels,
+                X_q0,
+                X_value_min,
+                X_value_max,
+                W_q0,
+                W_value_min,
+                W_value_max,
+            )
 
-        # Observe X_fp32 and determine X_scale and X_zero_point, this should match
-        # internals of dynamic linear.
-        X_scale, X_zp = _calculate_dynamic_qparams(X_fp32, torch.quint8)
-        X_q = torch.quantize_per_tensor(X_fp32, scale=X_scale, zero_point=X_zp, dtype=torch.quint8)
+            X_fp32 = torch.from_numpy(_dequantize(X_q0, X_scale, X_zp)).to(dtype=torch.float)
+            if use_multi_dim_input:
+                X_fp32 = X_fp32.view(3, int(batch_size / 3), input_channels)
 
-        # Weight prepacking operator for dynamic quantized Linear
-        W_prepack = qlinear_prepack(W_q, b_fp32)
-        # Dynamic quantized Linear operator with prepacked weight
-        Y_fp32 = qlinear_dynamic(X_q.dequantize(), W_prepack)
-        # Y_fp32 = qlinear_dynamic(X_fp32, W_prepack, b_fp32)
+            # W_scale, W_zp = _calculate_dynamic_qparams(W_fp32, torch.qint8)
+            # We currently only check the case where W_scale = 1.0, W_zp = 0.
 
-        Y_fp32_ref = F.linear(X_q.dequantize(), W_q.dequantize(), b_fp32)
-        # Y_fp32_ref = F.linear(X_fp32, W_fp32, b_fp32)
-        # if use_multi_dim_input:
-        #     Y_fp32_ref = Y_fp32_ref.view(3, int(batch_size / 3), output_channels)
+            if use_channelwise:
+                W_fp32 = torch.from_numpy(_dequantize(W_q0, W_scales.reshape(
+                    (-1, 1)), W_zps.reshape((-1, 1)))).to(dtype=torch.float)
+                W_q = torch.quantize_per_channel(W_fp32, scales=torch.from_numpy(W_scales),
+                                                 zero_points=torch.from_numpy(W_zps), axis=0, dtype=torch.qint8)
+                b_fp32 = torch.from_numpy(
+                    _dequantize(b_q0, X_scale * W_scales, 0)
+                ).to(dtype=torch.float) if use_bias else None
+            else:
+                W_fp32 = torch.from_numpy(_dequantize(
+                    W_q0, W_scales[0], W_zps[0])).to(dtype=torch.float)
+                W_q = torch.quantize_per_tensor(W_fp32, scale=W_scales[0], zero_point=(
+                    W_zps[0].astype(int).item()), dtype=torch.qint8)
+                b_fp32 = torch.from_numpy(
+                    _dequantize(b_q0, X_scale * int(W_scales[0].item()), 0)
+                ).to(dtype=torch.float) if use_bias else None
+
+            # Observe X_fp32 and determine X_scale and X_zero_point, this should match
+            # internals of dynamic linear.
+            X_scale, X_zp = _calculate_dynamic_qparams(X_fp32, torch.quint8)
```

Comment:
If we enable the `reduce_range` option in the future, we also need to do the corresponding changes to `_calculate_dynamic_qparams` (https://github.com/pytorch/pytorch/blob/f050b16dd95b2bcce9853882fd3fb07a6fd80378/torch/testing/_internal/common_quantized.py#L41) in order to facilitate the unit tests

Reply:
Good catch! I'll update this